main.py

```diff
@@ -0,0 +1,546 @@
+# The entrypoint for VxIngest
+# Responsible for checking the existing jobs documents and determining which need to be run
+
+import argparse
+import logging
+import os
+import shutil
+import sys
+import tarfile
+from datetime import datetime, timedelta
+from multiprocessing import Queue
+from pathlib import Path
+from typing import Callable, Optional, TypedDict
+
+import yaml
+from couchbase.auth import PasswordAuthenticator  # type: ignore
+from couchbase.cluster import Cluster  # type: ignore
+from couchbase.exceptions import CouchbaseException, TimeoutException  # type: ignore
+from couchbase.options import (  # type: ignore
+    ClusterOptions,
+    ClusterTimeoutOptions,
+    QueryOptions,
+)
+from prometheus_client import CollectorRegistry, Counter, Gauge, write_to_textfile
+
+import ctc_to_cb.run_ingest_threads
+import grib2_to_cb.run_ingest_threads
+import log_config
+import netcdf_to_cb.run_ingest_threads
+import partial_sums_to_cb.run_ingest_threads
+
+# Get a logger with this module's name to help with debugging
+logger = logging.getLogger(__name__)
+
+# Configure prometheus metrics
+# Note - we may need to import prometheus's multiprocessing libraries
+
+# Create a registry we can write to a file
+prom_registry = CollectorRegistry()
+
+# Use a gauge because we're doing 1 file per job run, a histogram could be more appropriate.
+# Note - if we used a historgram or summary, we could apply a decorator directly to the function we're interested in
+prom_duration = Gauge(
+    "run_ingest_duration",
+    "The duration of an ingest run, in seconds",
+    registry=prom_registry,
+)
+prom_successes = Counter(
+    "run_ingest_success_count",
+    "The number of successful ingest jobs",
+    registry=prom_registry,
+)
+prom_failures = Counter(
+    "run_ingest_failure_count",
+    "The number of failed ingest jobs",
+    registry=prom_registry,
+)
+prom_last_success = Gauge(
+    "job_last_success_unixtime",
+    "Last time a batch job successfully finished",
+    registry=prom_registry,
+)
+
+
+def process_cli():
+    """Processes the following flags
+
+    -c - credentials file path
+    -m - metrics directory path
+    -o - output directory path
+    -x - "transfer" directory path
+    -l - log directory path
+    -j - jobid (optional)
+    -s - start epoch (optional?)
+    -e - end epoch (optional?)
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-j",
+        "--job_id",
+        type=str,
+        required=False,
+        help="An optional couchbase job document id",
+    )
+    parser.add_argument(
+        "-c",
+        "--credentials_file",
```

Comment:
Where is config.yaml? Should the credentials not be required?

Reply:
`config.yaml` is assumed to be in your shell's current working directory when you call `main.py`.  It isn't required as `config.yaml` is the default value if the user doesn't otherwise specify a path with the `-c` variable. I tried to make it work similarly to how `.env` files are known to work. If `main.py` can't find `config.yaml` (or the Path it was passed) it will give the user some immediate useful error output. 

https://github.com/NOAA-GSL/VxIngest/blob/894e5d28e2e3434af8b617d33346c53de2ef4aa7/main.py#L142-L151